### PR TITLE
Add maintainer for deleting unused file references

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java [hunk 1]
```diff
@@ -269,52 +272,89 @@ public boolean remove(ApplicationId applicationId) {
...
+        Set<String> fileReferencesInUse = new HashSet<>();
+        Set<ApplicationId> applicationIds = listApplications();
+        applicationIds.forEach(applicationId -> fileReferencesInUse.addAll(getApplication(applicationId).getModel().fileReferences()
+                                           .stream()
+                                           .map(FileReference::value)
+                                           .collect(Collectors.toSet())));
+        log.log(LogLevel.INFO, "File references in use : " + fileReferencesInUse);
+
+        // Find those on disk that are not in use
+        if (!fileReferencesPath.isDirectory())
```
Comment (mpolden) [inline configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java]: Consider moving guard to the top of the method.

File: configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java [hunk 2]
```diff
@@ -110,17 +113,48 @@ public void decideVersion() {
         assertEquals(Vtag.currentVersion, ApplicationRepository.decideVersion(regularApp, Environment.perf, targetVersion));
     }
 
+    @Test
+    public void deleteUnusedFileReferences() {
+        File fileReferencesDir = Files.createTempDir();
```
Comment (mpolden) [inline configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: If this test fails before deletion (e.g. failing assert), the directory will be left on disk. This is likely the case for all uses of `Files.createTempDir()` below.

Consider using something like
```
@Rule
public TemporaryFolder folder = new TemporaryFolder();
```

File: configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java [hunk 3]
```diff
@@ -110,17 +113,48 @@ public void decideVersion() {
...
+        applicationRepository = new ApplicationRepository(tenantRepository, provisioner, clock);
+        timeoutBudget = new TimeoutBudget(clock, Duration.ofSeconds(60));
+
+        // TODO: Deploy an app with a bundle or file that will be a file reference, too much missing in test setup to get this working now
+        PrepareParams prepareParams = new PrepareParams.Builder().applicationId(applicationId()).ignoreValidationErrors(true).build();
+        deployApp(new File("src/test/apps/app"), prepareParams);
+
+        boolean deleteFiles = false;
+        Set<String> toBeDeleted = applicationRepository.deleteUnusedFiledistributionReferences(fileReferencesDir, deleteFiles);
+        assertEquals(new HashSet<>(Collections.singletonList("foo")), toBeDeleted);
```
Comment (mpolden) [inline configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: Can be simplified to `Collections.singleton("foo")`.

File: configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java [hunk 4]
```diff
@@ -110,17 +113,48 @@ public void decideVersion() {
...
+        assertTrue(filereferenceDir.exists());
+
+        deleteFiles = true;
+        toBeDeleted = applicationRepository.deleteUnusedFiledistributionReferences(fileReferencesDir, deleteFiles);
+        assertEquals(new HashSet<>(Collections.singletonList("foo")), toBeDeleted);
```
Comment (mpolden) [inline configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java]: Same as above.

**Comment by hmusum** [comment]
Thanks, I have done the changes you suggested, please take a look again.

**Comment by mpolden** [comment]
LGTM, but there are many other uses of `Files.createTempDir()` in this diff that likely should use `TemporaryFolder` as well.

**Comment by hmusum** [comment]
Yeah, some code was moved and I did not change how this was done. I think I have fixed the ones seen in this PR, please take a look again

**Pull request merged**